### DISCUSSION TITLE
[C128] – Is the “indirect y addressing” for “xxx (sp),y”  parsed additional or has dependencies ?

### DISCUSSION THREAD

**Discussion by RetrofanC** [cc65/cc65#1931] [Q&A] [answered]
As it seems that some of my modules are crashing around the code which is using one of the;  

-  lda (sp),y 
- sta (sp),y
- adc (sp),y

I am wondering if the indirect y addressing of the stack pointer goes through some additional parsing?
(Is there an additional module needed to do this indirect y addressing which is needed to be in the sames ram space perhaps or has a dependency ?) 

When I look at the actual code in the C128 monitor it is translated to "($02),y" which seem correct right? 
(Given the "sp" is on $02 and $03 according to mapfile.txt) 

(Assuming that this ‘sp’ is pointing at the cc65 software stack (starting at  $09FF in my case) and not at the hardware stack (starting at $1FA))

**Comment by mrdudz** [comment, marked as answer]
> I am wondering if the indirect y addressing of the stack pointer goes through some additional parsing?
> (Is there an additional module needed to do this indirect y addressing which is needed to be in the sames ram space perhaps or has a dependency ?)

I don't really understand the question - but i guess the answer is "no" :)

If it is crashing at an instruction like this (is it really?) then perhaps for some reason "sp" is pointing to some I/O registers that it shouldnt write to at this point? ie when it is doing that, the banking is wrong?

**Reply by RetrofanC** [reply]
Thanks for the ongoing support and suggestions, appreciated!

It now seems that it might have to do with a conflict in the A-Reg and not with a conflict in (sp) indirect addressing actually.
It looks when I am using __cdecl__ instead of default (which __fastcall__ right?) it works more stable and doesn't crash at those points.

So it can be that with the bank switching and A/X parameters passing the system gets say "too confused".

To be fair what happens is a function in main (RAM#0) is calling a function to be executed in a foreigen bank (RAM#3).
Then as part of that function running in RAM#3 a "printf" is done which has to jump back to RAM#0 to handle before continuing the function in RAM#3. 

When I have time I will dig out why it goes wrong more close but for now it seems to work sofar that I can continue for now.
